Update development docs to the current state for the rails app.

brandocalrissian opened the pull request:
Change the setup script to point to the existing .env.example.

Add SECRET_KEY_BASE to .env.example.

brandocalrissian committed:
Update development docs to the current state for the rails app.

Change the setup script to point to the existing .env.example.

Add SECRET_KEY_BASE to .env.example.

jessieay commented on CONTRIBUTING.md:
```diff
@@ -9,17 +9,21 @@
-0. Go to `localhost:9292` and enjoy!
+0. Add the CROMERR account credentials (CDX_USERNAME and CDX_PASSWORD) to `.env`.
+0. Run `rake secret` and set the output as the value for `SECRET_KEY_BASE` in `.env`.
+0. Run `rails s` to start the Rails application.
```
let's suggest `foreman start` instead so the Procfile runs

jessieay commented on CONTRIBUTING.md:
```diff
@@ -41,9 +45,9 @@
-- To build the static site, run `rake build`.
-- To build and serve the app, run `rake serve`.
-- To deploy to 18F's cloud, run `rake deploy`.
+- ~~To build the static site, run `rake build`.~~
```
should we just remove this?

jessieay commented on CONTRIBUTING.md:
```diff
@@ -9,17 +9,21 @@
+0. Add the CROMERR account credentials (CDX_USERNAME and CDX_PASSWORD) to `.env`.
+0. Run `rake secret` and set the output as the value for `SECRET_KEY_BASE` in `.env`.
+0. Run `rails s` to start the Rails application.
+0. Go to `http://localhost:3000` and enjoy!
```
foreman does not default to `localhost:3000` like the standard rails server. Perhaps we should add a foreman config for PORT so we can tell ppl where to do. Otherwise, seems like it assigns randomly (mine is at 5100) https://robots.thoughtbot.com/foreman-as-process-manager

jessieay commented:
these changes look great! thanks for doing this. a few small comments from me

jessieay commented on CONTRIBUTING.md:
```diff
@@ -9,17 +9,21 @@
 0. Start postgres if it isn't automatically running: `postgres -D /usr/local/var/postgres`
```
let's remove `Copy the secret.rb.template to secret.rb. See below CROMERR Signing` while we're here

jessieay commented:
other things I am seeing that we can remove while we're doing this:

this section (we are removing js from the app, so no client tests):

> Client tests use karma/mocha/chai:

these are done via setup script:

> Run bundle install to grab the required gems.
> Use rake db:create:all to create the databases.
> User rake db:migrate to transform the database structure to what is needed by the app. For test you will need to preface with env variables to indicate the environment: RACK_ENV=test rake db:migrate

brandocalrissian commented:
Do we therefore need to run the setup script anytime we'd want to update gems or the database schema?

jessieay commented:
you can still run those commands in a one off basis but they aren't needed
to get set up initially, which is what I think these instructions are for

On Wednesday, January 13, 2016, Brandon Kirby notifications@github.com
wrote:

> Do we therefore need to run the setup script anytime we'd want to update
> gems or the database schema?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/18F/e-manifest/pull/45#issuecomment-171342981.

brandocalrissian committed:
Update CONTRIBUTING doc per suggestions.

Rename "rake tasks" section to "one-off tasks" to become more of
a catch-all.

Reorder the Procfile lines to make the puma web server get the port
a user specifies on the command line or an environment file, should
the user go down such a path.

jessieay commented:
this looks great! merging
